### ISSUE TITLE
Slowdown in debug mode due to log-structured allocator backtraces

### ISSUE THREAD

**Issue by avikivity** [scylladb/scylla#8975]
This is a profile of `database_test`:

```
     9.44%  database_test_g  database_test_g            [.] __asan_memcpy
     8.45%  database_test_g  database_test_g            [.] __gnu_cxx::__ops::_Iter_pred<seastar::decorate(unsigned long)::$_0>::operator()<__gnu_cxx::__normal_iterator<seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > > >
     4.67%  database_test_g  database_test_g            [.] std::__fill_a1<char*, int>
     4.27%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > >::operator++
     3.74%  database_test_g  database_test_g            [.] seastar::decorate(unsigned long)::$_0::operator()
     2.55%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > >::operator*
     1.73%  database_test_g  database_test_g            [.] std::__find_if<__gnu_cxx::__normal_iterator<seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > >, __gnu_cxx::__ops::_Iter_pred<seastar::decorate(unsigned long)::$_0> >
     1.06%  database_test_g  libc-2.32.so               [.] __memmove_sse2_unaligned_erms
     0.92%  database_test_g  database_test_g            [.] __asan_memset
     0.70%  database_test_g  database_test_g            [.] seastar::execution_stage::flush
     0.68%  database_test_g  database_test_g            [.] seastar::internal::try_reap_events
     0.65%  database_test_g  libc-2.32.so               [.] __memset_avx2_unaligned_erms
     0.63%  database_test_g  database_test_g            [.] __sanitizer::StackDepotBase<__sanitizer::StackDepotNode, 1, 20>::Put
     0.63%  database_test_g  database_test_g            [.] __asan::Allocator::Allocate
     0.53%  database_test_g  database_test_g            [.] seastar::internal::usable
     0.53%  reactor-1        database_test_g            [.] std::__fill_a1<char*, int>
     0.51%  database_test_g  database_test_g            [.] std::vector<seastar::shared_object, std::allocator<seastar::shared_object> >::begin
     0.49%  database_test_g  database_test_g            [.] __sanitizer::internal_memmove
     0.48%  database_test_g  database_test_g            [.] seastar::reactor::poll_once
     0.46%  database_test_g  database_test_g            [.] __sanitizer::BufferedStackTrace::UnwindFast
     0.44%  database_test_g  database_test_g            [.] seastar::decorate
     0.43%  database_test_g  database_test_g            [.] seastar::internal::execution_stage_manager::flush
     0.43%  database_test_g  database_test_g            [.] std::vector<seastar::shared_object, std::allocator<seastar::shared_object> >::end
     0.37%  database_test_g  database_test_g            [.] __asan::QuarantineCallback::Recycle
     0.37%  reactor-1        database_test_g            [.] __asan_memcpy
     0.37%  database_test_g  database_test_g            [.] __asan::AsanChunk::UsedSize
     0.37%  database_test_g  database_test_g            [.] __gnu_cxx::operator==<seastar::execution_stage**, seastar::execution_stage**, std::vector<seastar::execution_stage*, std::allocator<seastar::execution_stage*> > >
     0.32%  database_test_g  database_test_g            [.] seastar::current_backtrace_tasklocal()::$_3::operator()
     0.32%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::execution_stage**, std::vector<seastar::execution_stage*, std::allocator<seastar::execution_stage*> > >::base
     0.31%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::execution_stage**, std::vector<seastar::execution_stage*, std::allocator<seastar::execution_stage*> > >::operator++
     0.31%  database_test_g  database_test_g            [.] __interceptor_memcpy.part.0
     0.29%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::pollfn**, std::vector<seastar::pollfn*, std::allocator<seastar::pollfn*> > >::base
     0.29%  database_test_g  database_test_g            [.] __gnu_cxx::operator==<seastar::pollfn**, seastar::pollfn**, std::vector<seastar::pollfn*, std::allocator<seastar::pollfn*> > >
     0.29%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > >::base
     0.27%  database_test_g  database_test_g            [.] std::basic_string_view<signed char, std::char_traits<signed char> >::remove_prefix
     0.27%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::pollfn**, std::vector<seastar::pollfn*, std::allocator<seastar::pollfn*> > >::operator++
     0.26%  database_test_g  database_test_g            [.] boost::container::vector<seastar::frame, boost::container::dtl::static_storage_allocator<seastar::frame, 64ul, 0ul, true>, void>::emplace_back<seastar::frame>
     0.26%  database_test_g  database_test_g            [.] managed_bytes_basic_view<(mutable_view)0>::managed_bytes_basic_view
     0.26%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > >::__normal_iterator
     0.25%  database_test_g  database_test_g            [.] std::find_if<__gnu_cxx::__normal_iterator<seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > >, seastar::decorate(unsigned long)::$_0>
     0.23%  database_test_g  database_test_g            [.] std::__find_if<__gnu_cxx::__normal_iterator<seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > >, __gnu_cxx::__ops::_Iter_pred<seastar::decorate(unsigned long)::$_0> >
     0.23%  database_test_g  database_test_g            [.] std::this_thread::get_id
     0.23%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<dl_phdr_info*, std::vector<dl_phdr_info, std::allocator<dl_phdr_info> > >::base
     0.23%  database_test_g  database_test_g            [.] std::basic_string_view<signed char, std::char_traits<signed char> >::substr
     0.23%  database_test_g  database_test_g            [.] __interceptor_strlen.part.0
     0.23%  database_test_g  database_test_g            [.] seastar::reactor::run_tasks
     0.22%  database_test_g  database_test_g            [.] seastar::reactor::run_some_tasks
     0.22%  database_test_g  libpthread-2.32.so         [.] __pthread_getspecific
     0.22%  database_test_g  database_test_g            [.] dl_iterate_phdr
     0.21%  database_test_g  database_test_g            [.] __gnu_cxx::operator-<seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > >
     0.21%  database_test_g  database_test_g            [.] seastar::reactor::account_runtime
     0.20%  database_test_g  database_test_g            [.] std::vector<dl_phdr_info, std::allocator<dl_phdr_info> >::begin
     0.20%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<dl_phdr_info*, std::vector<dl_phdr_info, std::allocator<dl_phdr_info> > >::operator++
     0.20%  database_test_g  libgcc_s-10-20201125.so.1  [.] 0x0000000000012b9a
     0.20%  database_test_g  database_test_g            [.] __gnu_cxx::operator==<dl_phdr_info*, dl_phdr_info*, std::vector<dl_phdr_info, std::allocator<dl_phdr_info> > >
     0.20%  database_test_g  database_test_g            [.] std::basic_string_view<signed char, std::char_traits<signed char> >::size
     0.20%  database_test_g  database_test_g            [.] managed_bytes_basic_view<(mutable_view)0>::remove_prefix
     0.20%  database_test_g  database_test_g            [.] __asan::QuickCheckForUnpoisonedRegion
     0.19%  database_test_g  database_test_g            [.] __asan_memmove
     0.19%  database_test_g  [kernel.kallsyms]          [k] syscall_exit_to_user_mode
     0.19%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::pollfn**, std::vector<seastar::pollfn*, std::allocator<seastar::pollfn*> > >::operator*
     0.19%  database_test_g  database_test_g            [.] __gnu_cxx::operator==<seastar::shared_object const*, seastar::shared_object const*, std::vector<seastar::shared_object, std::allocator<seastar::shared_object> > >
     0.18%  database_test_g  libgcc_s-10-20201125.so.1  [.] 0x00000000000127cc
     0.18%  database_test_g  database_test_g            [.] seastar::backtrace<seastar::current_backtrace_tasklocal()::$_3>
     0.18%  database_test_g  database_test_g            [.] compound_type<(allow_prefixes)1>::iterator::read_current
     0.18%  database_test_g  database_test_g            [.] std::vector<seastar::shared_object, std::allocator<seastar::shared_object> >::empty
     0.17%  database_test_g  database_test_g            [.] __gnu_cxx::__normal_iterator<seastar::execution_stage**, std::vector<seastar::execution_stage*, std::allocator<seastar::execution_stage*> > >::operator*
     0.17%  database_test_g  libc-2.32.so               [.] memcpy@GLIBC_2.2.5
     0.17%  database_test_g  database_test_g            [.] __asan_region_is_poisoned
     0.16%  database_test_g  database_test_g            [.] managed_bytes_basic_view<(mutable_view)0>::remove_current
     0.16%  database_test_g  database_test_g            [.] __asan::PoisonShadow
```

Many of the functions are related to backtrace collection - seastar::decorate, anything with seastar::shared_object, dl_phdr_info, and more. Some of the generic asan overhead (__asan_memcpy) can also be attributed to this functionality. But in a successful run, these backtraces are just thrown away, so they only slow down the test without contributing anything useful.

**Comment by avikivity** [comment]
This functionality should be made optional and non-default.

**Comment by avikivity** [comment]
Fixed by 4c01a88c9d020109a65b715d89aa9a31b4c141e6
